src/ophys_etl/modules/decrosstalk/qc_plotting.py

```diff
@@ -0,0 +1,1147 @@
+            local_m = qc_data[f'ROI/{roi}/roi/unmixed/mixing_matrix'][()]
+            if not qc_data[f'ROI/{roi}/roi/unmixed/converged'][()]:
+                return local_m
+            avg_matrix += local_m
+            n_m += 1
+    return avg_matrix/n_m
+
+
+def plot_plane_pair(ophys_planes: Tuple[DecrosstalkingOphysPlane,
+                                        DecrosstalkingOphysPlane],
+                    roi_flags: dict,
+                    subplot_spec: matplotlib.gridspec.SubplotSpec,
+                    fig) -> None:
+    """
+    Plot a single pair of Ophys planes in the larger plot created
+    by generate_roi_figure
+
+    Parameters
+    ----------
+    ophys_planes: Tuples[DecrosstalkingOphysPlane,
+                         DecrosstalkingOphysPlane]
+        The two coupled planes to be plotted
+
+    roi_flags: dict
+       A dict mapping roi_id to flags ('invalid' or 'ghost')
+
+    subplot_spec: matplotlib.gridspec.SubplotSpec
+        The section of the outer grid in which this pair of
+        planes will be plotted
+
+    fig: matplotlib.figure.Figure
+        The figure in which we are plotting
+
+    Returns
+    -------
+    None
+    """
+
+    green = matplotlib.colors.to_rgba('green')
+    green = tuple([int(255*g) for g in green[:3]])
+
+    red = matplotlib.colors.to_rgba('red')
+    red = tuple([int(255*r) for r in red[:3]])
+
+    outlier_color = (255, 108, 220)
+    valid_color = (255, 196, 92)
+
+    outlier_color_hex = '#%02x%02x%02x' % outlier_color
+    valid_color_hex = '#%02x%02x%02x' % valid_color
+    green_hex = '#%02x%02x%02x' % green
+    red_hex = '#%02x%02x%02x' % red
+
+    inner_grid = gridspec.GridSpecFromSubplotSpec(2, 4,
+                                  subplot_spec=subplot_spec,    # noqa E128
+                                  wspace=0.05, hspace=0.)       # noqa E128
+
+    # loop over the two planes
+    for ii in range(2):
+        plane = ophys_planes[ii]
+
+        with h5py.File(plane.qc_file_path, 'r') as in_file:
+            if 'ROI' not in in_file:
+                # There were no ROIs in this plane; just move on
+                continue
+            roi_keys = list(in_file['ROI'].keys())
+
+        roi_id = np.array([int(k) for k in roi_keys])
+        roi_min = roi_id.min()
+
+        avg_mixing_matrix = get_avg_mixing_matrix(plane)
+        roi_qc = find_problematic_rois(plane, roi_flags)
+        n_valid_roi = 0
+        n_ghost_roi = 0
+
+        # load the raw max projection image
+        raw_img = PIL.Image.open(plane.maximum_projection_image_path)
+        n_rows = raw_img.size[0]
+        n_cols = raw_img.size[1]
+        raw_img = np.array(raw_img).reshape(n_rows, n_cols)
+
+        # convert to an RGBA image
+        max_img = np.zeros((n_rows, n_cols, 4), dtype=int)
+
+        for jj in range(3):
+            max_img[:, :, jj] = raw_img[:, :]
+        max_img[:, :, 3] = 255
+
+        # superimpose gridlines
+        max_img = add_gridlines(max_img, 3)
```

Comment:
this seems a little funky. I would have offloaded the gridlines to the matplotlib axes, using `ax.axvline()`
or something. You might need a small function to tell you the locations, but, then you could control the look of them without trying to modify image pixels.

Reply:
I'm not sure how committed I am to this position, but the advantage I see is that, given we are plotting every image multiple times, by adding the gridlines directly to the max projection image, we don't have to remember to call vline and hline every time we run imshow. As long as we are plotting a copy of the image to which we added the gridlines, we get the grid lines "for free."

Really, I guess it's just a matter of personal preference whether one thinks it is more annoying to alter the image pixels or to do the nested loops needed to get the gridlines on every instance of the maximum projection plane.

I guess I don't feel that strongly. If the above doesn't convince you, I'll change it.

Reply:
it's fine to leave it. if these plots were interactive, it might be more weird.